pkg/graphql/request_test.go

```diff
@@ -0,0 +1,43 @@
+package graphql
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnmarshalRequest(t *testing.T) {
+	t.Run("should return error when request is empty", func(t *testing.T) {
+		requestBytes := []byte("")
+		requestBuffer := bytes.NewBuffer(requestBytes)
+
+		request, err := UnmarshalRequest(requestBuffer)
+
+		assert.Error(t, err)
+		assert.Equal(t, ErrEmptyRequest, err)
+		assert.Nil(t, request)
+	})
+
+	t.Run("should return error when query is empty", func(t *testing.T) {
+		requestBytes := []byte("{\"query\": \"\"}")
+		requestBuffer := bytes.NewBuffer(requestBytes)
+
+		request, err := UnmarshalRequest(requestBuffer)
+
+		assert.Error(t, err)
+		assert.Equal(t, ErrEmptyRequest, err)
+		assert.Nil(t, request)
+	})
+
+	t.Run("should successfully unmarshal request", func(t *testing.T) {
+		requestBytes := []byte("{\"operation_name\": \"Hello\", \"variables\": \"\", \"query\": \"query Hello { hello }\"}")
```

Comment:
Better to use `` like `{"operation_name": "Hello"}` instead of putting backslashes